### ISSUE TITLE
Rule proposal: Error when adding a class that begins with a dot`classList.add('.`

### ISSUE THREAD

**Issue by universemaster** [sindresorhus/eslint-plugin-unicorn#1837]
### Description

Sometimes, we incorrectly try to add a class that begins with a dot, but linters and indeed browsers do not indicate that this is an error.

The error occurs because we use a dot in selectors for classname, but the dot is not part of the actual class name.

### Fail

```js
node.classList.add('.table-of-contents');
```

### Pass

```js
node.classList.add('table-of-contents')
```

**Comment by sindresorhus** [comment]
It should apply to the other `classList` methods too.

**Comment by sindresorhus** [comment]
Accepted

**Comment by sindresorhus** [comment]
Please suggest a rule name.

**Comment by fregante** [comment]
`no-selector-as-class-name`. The rule could cover `element.id = '#id'` too but it's harder to properly report without types.

Alternatively it could generically target CSS identifiers that need to be escaped, so:

- don't use `classList.add('hello\world')` because it's a single class containing a slash, targetable as `.hello\\world {}` 


So maybe just `no-escapable-css-identifiers`? 😬
